### PR TITLE
pkg/ebpf: created events documentation

### PR DESCRIPTION
**Pull request by AlonZivony**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [ ] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Created field in event's definition for documentation.
Added magic_write event doc for example of how events documentations
should look like.

Fixes: #1398 

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [ ] My code follows the style guidelines (C and Go) of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [ ] Subject starts with "subsystem|file: description".
- [ ] Do not end the subject line with a period.
- [ ] Limit the subject line to 50 characters.
- [ ] Separate subject from body with a blank line.
- [ ] Use the imperative mood in the subject line.
- [ ] Wrap the body at 72 characters.
- [ ] Use the body to explain what and why instead of how.

### CODE REVIEW

**Comment by AlonZivony** [comment]
@rafaeldtinoco WDYT?

**Comment by yanivagman** [comment]
That's a good starting point to document tracee's events.

It would be nice to add an option to print the event's help from the cli, e.g. with:
`./tracee-ebpf --help magic_write`

**Comment by rafaeldtinoco** [comment]
> That's a good starting point to document tracee's events.
> 
> It would be nice to add an option to print the event's help from the cli, e.g. with: `./tracee-ebpf --help magic_write`

Hum, yep I like the cli idea. We can convert MD to GROFF and display the "just generated" man page out of that cli command: https://github.com/cpuguy83/go-md2man

File: docs/events/format.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+## Intro
+<event_name> - one sentence description of the event
+
+## Description
+Detailed description of the event. Should include:
+* What is the purpose of the event?
+* Are there any edge-cases, drawbacks or advantages of using it?
+
+## Arguments
+* `<arg#1>`[<tags>] - short description of the argument value.
```
Comment (AlonZivony) [inline docs/events/format.md:12]: Add argument type

Comment (AlonZivony) [reply docs/events/format.md:12]: Should elaborate on possible changes to the argument's value and type (like with arg-parser)

File: docs/events/security_alerts/magic_write.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+type and other meta-data needed to understand if the file is a threat.
+
+### Note
+The event doesn't occur for FIFO files or other files with no persistent offsets,
+to reduce spam events.
+
+## Arguments
+* `pathname`[K] - the path of the file written.
+* `bytes`[U,TOCTOU] - the first 20 bytes of the file.
+* `dev_t`[K] - the device the file resides in.
```
Comment (AlonZivony) [inline docs/events/security_alerts/magic_write.md:19]: This is the argument type, not name

File: pkg/ebpf/events_definitions.go [hunk 3]
```diff
@@ -5011,6 +5012,7 @@ var EventsDefinitions = map[int32]EventDefinition{
 	MagicWriteEventID: {
 		ID32Bit: sys32undefined,
 		Name:    "magic_write",
+		DocPath: "docs/events/security_alerts/magic_write.md",
```
Comment (AlonZivony) [inline pkg/ebpf/events_definitions.go:5015]: Path should be relative to "docs/events/"

**Comment by AlonZivony** [comment]
@yanivagman or @rafaeldtinoco if one of you can approve this PR, I think it is ready for merge.
I think that from the moment you approve it we need to make it mandatory for every new event to apply it.
Maybe @rafaeldtinoco you should put it in the PR checklist.

File: docs/events/security_alerts/magic_write.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# magic_write
+
+## Intro
+magic_write - write operation to a file which changed the file's headers
+
+## Description
+An event marking that a new file is written, or an existing file type changed.
```
Comment (yanivagman) [inline docs/events/security_alerts/magic_write.md:7]: Not necessarily that the file type is changed, but the header of the file is.

**Review by yanivagman — APPROVED** [review]
I'm +1 here but let's wait for @rafaeldtinoco to review as well, and maybe add this to the CI pipeline

**Comment by rafaeldtinoco** [comment]
> I'm +1 here but let's wait for @rafaeldtinoco to review as well, and maybe add this to the CI pipeline

Im +1 here as well. Thanks! And, I think we should enforce:

image

all added events, from now on, should have a DocPath and a markdown describing it in the correct format (that is what we should enforce, right ?).

**Comment by AlonZivony** [comment]
> all added events, from now on, should have a DocPath and a markdown describing it in the correct format (that is what we should enforce, right ?).

Yea, we should enforce the `DocPath` existence and the file format.
I will update the PR according to Yaniv's request and you will merge afterwards.

**Comment by AlonZivony** [comment]
@rafaeldtinoco I think you can merge the PR now

**Pull request merged**